pkg/hubble/filters/tcp_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2020 Authors of Hubble
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package filters
+
+import (
+	"context"
+	"testing"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+)
+
+func TestFlowTCPFilter(t *testing.T) {
+	type args struct {
+		f  []*flowpb.FlowFilter
+		ev *v1.Event
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+		want    bool
+	}{
+		{
+			// Filter set to SYN-ACK, received event SYN-ACK: Verdict: Allow
+			name: "filterSYNACK_eventSYNACK",
+			args: args{
+				f: []*flowpb.FlowFilter{{TcpFlags: []*flowpb.TCPFlags{
+					{SYN: true, ACK: true}}}},
+				ev: &v1.Event{Event: &flowpb.Flow{
+					L4: &flowpb.Layer4{Protocol: &flowpb.Layer4_TCP{TCP: &flowpb.TCP{
+						Flags: &flowpb.TCPFlags{SYN: true, ACK: true}}}},
+				}},
+			},
+			want: true,
+		},
```

Comment:
These test setup are very difficult to read. Since the only data that changes case to case is the tcp flags in the filter and the tcp flags in the event, you should change the inputs to be just those two slices. Then have a helper compose a filter and event before running through the test case.

Reply:
Agree. Will change this in the next commit.

Reply:
Changed the driver code and now it looks much better. Please have a look and thanks for the review.